Blazor RC2 Updates 8.0 (#30576)
dotnet/AspNetCore.Docs: aspnetcore/blazor/fundamentals/routing.md

Navigation.Refresh(true);
```

Enhanced navigation is enabled by default, but it can be controlled hierarchically and on a per-link basis using the `data-enhance-nav` HTML attribute.

The following examples disable enhanced navigation:

```html
<a href="redirect" data-enhance-nav="false">
    GET without enhanced navigation
</a>
```

```razor
<ul data-enhance-nav="false">
    <li>
        <a href="redirect">GET without enhanced navigation</a>
    </li>
    <li>
        <a href="redirect-2">GET without enhanced navigation</a>
    </li>
</ul>
```

If the destination is a non-Blazor endpoint, enhanced navigation doesn't apply, and the client-side JavaScript retries as a full page load. This ensures no confusion to the framework about external pages that shouldn't be patched into an existing page.

> Form POST requests adopt [enhanced navigation](xref:blazor/fundamentals/routing#enhanced-navigation-and-form-handling) for form POST requests with the *explicit addition* of the `Enhance` parameter to `EditForm` forms or the *explicit addition* of the `data-enhance` attribute to the HTML form (`<form>`):

Reviewer: Is the "explicit addition" text trying to say that the `data-enhance` attribute can't be added dynamically, like with JS? If yes, should we just explicitly state that as an additional note?

```suggestion
To enable enhanced form handling, add the `Enhance` parameter to `EditForm` forms or the `data-enhance` attribute to HTML forms (`<form>`):
```

```razor
<EditForm Enhance ...>
    ...
</EditForm>
```

```html
<form ... data-enhance>
    ...
</form>
```

> Enhanced navigation isn't hierarchical and doesn't flow to child forms:

Reviewer: ```suggestion
Enhanced form handling isn't hierarchical and doesn't flow to child forms:
```

<span aria-hidden="true">❌</span><span class="visually-hidden">Unsupported:</span> You can't set enhanced navigation on a form's ancestor element to enable enhanced navigation for the form.

```html
<div data-enhance>
    <form ...>
        <!-- NOT enhanced -->
    </form>
</div>
```

> [!WARNING]
> ***Don't try to enhance form navigation for forms that POST to non-Blazor endpoints.***
>

> > This is the main reason why enabling form POSTs with enhanced navigation requires an explicit gesture on each `EditForm` and `<form>` element and it doesn't flow down elements or to child forms. It isn't safe to assume that all forms in general should use enhanced posts because any given form might not target a Blazor endpoint.

Reviewer: See my comment on how to tone done this warning in the forms doc.
